src/components/PlaylistTable.test.jsx

```diff
@@ -87,8 +86,88 @@ describe("single playlist exporting", () => {
       expect(saveAsMock).toHaveBeenCalledWith(
         {
           content: [
-            '"Track URI","Track Name","Artist URI","Artist Name","Album URI","Album Name","Disc Number","Track Number","Track Duration (ms)","Added By","Added At"\n' +
-            '"spotify:track:1GrLfs4TEvAZ86HVzXHchS","Crying","spotify:artist:4TXdHyuAOl3rAOFmZ6MeKz","Six by Seven","spotify:album:4iwv7b8gDPKztLkKCbWyhi","Best of Six By Seven","1","3","198093","","2020-07-19T09:24:39Z"\n'
+            '"Track URI","Track Name","Artist URI","Artist Name","Album URI","Album Name","Album Release Date","Disc Number","Track Number","Track Duration (ms)","Explicit","Popularity","Added By","Added At"\n' +
+            '"spotify:track:1GrLfs4TEvAZ86HVzXHchS","Crying","spotify:artist:4TXdHyuAOl3rAOFmZ6MeKz","Six by Seven","spotify:album:4iwv7b8gDPKztLkKCbWyhi","Best of Six By Seven","2017-02-17","1","3","198093","false","2","","2020-07-19T09:24:39Z"\n'
+          ],
+          options: { type: 'text/csv;charset=utf-8' }
+        },
+        'liked.csv',
+        true
+      )
+    })
+  })
+
+  test("including additional artist data", async () => {
```

Comment:
Can you elaborate for me what this testing framework is about and how to use it?

Reply:
I'm actually going to write something (possibly a blog article) on working with the specifics of working with [Jest](https://jestjs.io/en/) and [MSW](https://mswjs.io/) for the first time. Overally it's been great though with a few caveats.

I'm not sure if that's what you were asking or if you were asking a more general question. Just in case, in a more general sense, the test suite is there to validate the behavior of the app. It's mostly about guarding against regressions so I'm free to refactor without worrying too much about breaking things.

In this particular case, most of my tests take the form of mocking out the API calls as close to the HTTP transport layer as possible (so as to be client-agnostic), and then asserting that the app behaves in the expected way given a particular set of responses.

The tests don't actually call the API, so it wouldn't help in the case of breaking changes to the Spotify API. For that I have to rely on user feedback but I also have a last line of defence: exception monitoring with Bugsnag.